newspaper/network.py

```diff
@@ -31,44 +33,52 @@ def get_request_kwargs(timeout, useragent):
 
 
 def get_html(url, config=None, response=None):
-    """Retrieves the html for either a url or a response object. All html
-    extractions MUST come from this method due to some intricies in the
-    requests module. To get the encoding, requests only uses the HTTP header
-    encoding declaration requests.utils.get_encoding_from_headers() and reverts
-    to ISO-8859-1 if it doesn't find one. This results in incorrect character
-    encoding in a lot of cases.
+    """HTTP response code agnostic 
+    """
+    try:
+        return get_html_2XX_only(url, config, response)
+    except requests.exceptions.RequestException as e:
+        log.debug('get_html() error. %s on URL: %s' % (e, url))
+        return ''
+
+
+def get_html_2XX_only(url, config=None, response=None):
+    """Consolidated logic for http requests from newspaper. We handle error cases:
+    - Attempt to find encoding of the html by using HTTP header. Fallback to 
+      'ISO-8859-1' if not provided.
+    - Error out if a non 2XX HTTP response code is returned.
     """
-    FAIL_ENCODING = 'ISO-8859-1'
     config = config or Configuration()
     useragent = config.browser_user_agent
     timeout = config.request_timeout
 
     if response is not None:
-        if response.encoding != FAIL_ENCODING:
-            return response.text
-        return response.content
+        return _get_html_from_response(response)
 
     try:
-        html = None
-
         response = requests.get(
             url=url, **get_request_kwargs(timeout, useragent))
+    except requests.exceptions.RequestException as e:
```

Comment:
This is important. STILL catch `requests` exceptions in this function, but DON'T CATCH the 4XX or 5XX response codes. This disambiguation will allow our callers to have the HTTPException when a 4XX or 5XX happens and act accordingly